### DISCUSSION TITLE
Neue Anlage (2024.09.11): keine Kommunikation mit Victron VE.Direct ==> ERROR: debug buffer overrun!

### DISCUSSION THREAD

**Discussion by Matti07** [hoylabs/OpenDTU-OnBattery#1275] [Q&A] [answered]
Moin,
meine neue Anlage (Fusion V2 mit CAN / Iso Shield) will nicht mit dem Victron sprechen. Außer
"[VE.Direct MPPT 11/12] ERROR: debug buffer overrun!"  kommt da nichts.

11:36:49.623 > RX Period End
11:36:49.643 > Success
11:36:49.664 > Fetch inverter: 116493102514
11:36:49.670 > [DPL::updateInverter] actual limit is 14.1 % (282 W respectively), effective 2041 ms after update started, requested were 14.1 %
11:36:49.675 > [DPL::announceStatus] waiting for sufficiently recent inverter data
11:36:49.681 > TX RealTimeRunData 865.00 MHz --> 15 93 10 25 14 80 16 89 92 80 0B 00 66 F1 36 B1 00 00 00 00 00 00 00 00 0A F6 4D 
11:36:49.699 > Interrupt received
11:36:49.707 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 01 00 01 02 08 02 09 00 3B 00 41 01 33 01 52 00 00 B0 | 2 dBm
11:36:49.723 > Interrupt received
11:36:49.728 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 02 03 B5 00 00 03 D6 00 00 00 00 02 09 02 09 00 40 8B | 2 dBm
11:36:49.747 > Interrupt received
11:36:49.752 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 03 00 3E 01 51 01 45 00 00 03 CF 00 00 03 DB 00 00 97 | 2 dBm
11:36:49.788 > Interrupt received
11:36:49.791 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 04 00 00 09 3D 13 89 04 DC 00 00 00 34 03 E8 00 A6 A1 | 2 dBm
11:36:49.828 > Interrupt received
11:36:49.833 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 85 00 03 55 BE C7 | 2 dBm
11:36:50.084 > Websocket: [/livedata][2] disconnect
11:36:50.119 > Websocket: [/vedirectlivedata][2] disconnect
11:36:50.124 > Websocket: [/batterylivedata][2] disconnect
11:36:50.128 > [Pylontech] Received CAN message: 0x0359 - 00 00 00 00 01 50 4E
11:36:50.147 > [Pylontech] Alarms: 0 0 0 0 0 0 0
11:36:50.167 > [Pylontech] Warnings: 0 0 0 0 0 0 0
11:36:50.171 > [Pylontech] Received CAN message: 0x0351 - 14 02 72 01 72 01 C2 01
11:36:50.177 > [Pylontech] chargeVoltage: 53.200001 chargeCurrentLimitation: 37.000000 dischargeCurrentLimitation: 37.000000
11:36:50.203 > RX Period End
11:36:50.223 > Success
11:36:50.231 > [DPL::announceStatus] waiting for sufficiently recent power meter reading
11:36:50.236 > [Pylontech] Received CAN message: 0x0355 - 52 00 64 00
11:36:50.260 > [Pylontech] soc: 82.000000 soh: 100
11:36:50.269 > [Pylontech] Received CAN message: 0x0356 - 07 14 EC 00 BC 00
11:36:50.272 > [Pylontech] voltage: 51.270000 current: 23.600000 temperature: 18.800001
11:36:50.276 > [Pylontech] Received CAN message: 0x035C - C0 00
11:36:50.296 > [Pylontech] chargeStatusBits: 1 1 0
11:36:50.301 > [Pylontech] Received CAN message: 0x035E - 50 59 4C 4F 4E 20 20 20
11:36:50.321 > [Pylontech] Manufacturer: PYLON   
11:36:50.597 > [PowerMeterHttpJson] New total: 39.72
11:36:51.518 > [PowerMeterHttpJson] New total: 2115.68
11:36:51.553 > Fetch inverter: 116493102514
11:36:51.579 > TX RealTimeRunData 865.00 MHz --> 15 93 10 25 14 80 16 89 92 80 0B 00 66 F1 36 B3 00 00 00 00 00 00 00 00 6A EF 36 
11:36:51.639 > Interrupt received
11:36:51.646 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 01 00 01 02 07 02 09 00 6A 00 71 02 25 02 4D 00 00 D7 | 2 dBm
11:36:51.687 > Interrupt received
11:36:51.694 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 02 03 B5 00 00 03 D6 00 00 00 00 02 08 02 08 00 71 BA | 2 dBm
11:36:51.739 > Interrupt received
11:36:51.744 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 03 00 6F 02 4C 02 40 00 00 03 CF 00 00 03 DB 00 00 DE | 2 dBm
11:36:51.787 > Interrupt received
11:36:51.793 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 04 00 00 09 3F 13 89 08 8C 00 00 00 5C 03 E8 00 A6 97 | 2 dBm
11:36:51.827 > Interrupt received
11:36:51.834 > RX 865.00 MHz --> 95 93 10 25 14 80 16 89 92 85 00 03 6E 03 41 | 2 dBm
11:36:52.132 > [Pylontech] Received CAN message: 0x0359 - 00 00 00 00 01 50 4E
11:36:52.196 > [Pylontech] Alarms: 0 0 0 0 0 0 0
11:36:52.199 > [Pylontech] Warnings: 0 0 0 0 0 0 0
11:36:52.205 > [Pylontech] Received CAN message: 0x0351 - 14 02 72 01 72 01 C2 01
11:36:52.209 > [Pylontech] chargeVoltage: 53.200001 chargeCurrentLimitation: 37.000000 dischargeCurrentLimitation: 37.000000
11:36:52.231 > RX Period End
11:36:52.251 > Success
11:36:52.257 > [Pylontech] Received CAN message: 0x0355 - 52 00 64 00
11:36:52.274 > [Pylontech] soc: 82.000000 soh: 100
11:36:52.279 > [Pylontech] Received CAN message: 0x0356 - 03 14 D7 00 BC 00
11:36:52.285 > [Pylontech] voltage: 51.230000 current: 21.500000 temperature: 18.800001
11:36:52.291 > [Pylontech] Received CAN message: 0x035C - C0 00
11:36:52.311 > [Pylontech] chargeStatusBits: 1 1 0
11:36:52.318 > [Pylontech] Received CAN message: 0x035E - 50 59 4C 4F 4E 20 20 20
11:36:52.339 > [Pylontech] Manufacturer: PYLON   
11:36:52.344 > [VE.Direct MPPT 11/12] ERROR: debug buffer overrun!
11:36:52.542 > [PowerMeterHttpJson] New total: 2193.13
11:36:52.549 > [DPL::loop] ******************* ENTER **********************
11:36:52.557 > [DPL::loop] battery interface enabled, SoC: 82.000000 %, StartTH: 80 %, StopTH: 12 %, SoC age: 0 s, ignore: no
11:36:52.560 > [DPL::getBatteryVoltage] BMS: 51.23 V, MPPT: -1.00 V, inverter: 51.90 V, returning: 51.23V
11:36:52.568 > [DPL::loop] dcVoltage: 51.23 V, loadCorrectedVoltage: 51.45 V, StartTH: 47.00 V, StopTH: 45.00 V
11:36:52.571 > [DPL::loop] StartTH reached: yes, StopTH reached: no, SolarPT disabled, use at night: yes
11:36:52.575 > [DPL::calcPowerLimit] battery use allowed, solar power (DC): 0 W
11:36:52.581 > [DPL::calcPowerLimit] target consumption: -1 W, base load: 95 W, power meter does include inverter output
11:36:52.588 > [DPL::calcPowerLimit] power meter value: 2193 W, power meter valid: yes, inverter output: 218 W, solar power (AC): 0 W
11:36:52.596 > [DPL::calcPowerLimit] match household consumption with limit of 2412 W
11:36:52.602 > [DPL::setNewPowerLimit] input limit: 2412 W, min limit: 40 W, max limit: 1650 W, hysteresis: 10 W
11:36:52.607 > [DPL::setNewPowerLimit] inverter max: 2000 W, inverter is producing, requesting: 1650 W, reported: 282 W, diff: 1368 W
11:36:52.612 > [DPL::updateInverter] sending limit of 82.5 % (1650 W respectively), max output is 2000 W
11:36:52.617 > TX ActivePowerControl 865.00 MHz --> 51 93 10 25 14 80 16 89 92 81 0B 00 03 39 00 01 E9 10 26 
11:36:52.623 > [DPL::announceStatus] waiting for a power limit command to complete
11:36:52.643 > Interrupt received
11:36:52.647 > RX 865.00 MHz --> D1 93 10 25 14 80 16 89 92 81 00 00 0B 00 14 07 77 | 2 dBm
11:36:53.499 > [PowerMeterHttpJson] New total: 2168.67
11:36:54.181 > [Pylontech] Received CAN message: 0x0359 - 00 00 00 00 01 50 4E
11:36:54.228 > [Pylontech] Alarms: 0 0 0 0 0 0 0
11:36:54.247 > [Pylontech] Warnings: 0 0 0 0 0 0 0
11:36:54.252 > [Pylontech] Received CAN message: 0x0351 - 14 02 72 01 72 01 C2 01
11:36:54.259 > [Pylontech] chargeVoltage: 53.200001 chargeCurrentLimitation: 37.000000 dischargeCurrentLimitation: 37.000000
11:36:54.264 > [Pylontech] Received CAN message: 0x0355 - 52 00 64 00
11:36:54.284 > [Pylontech] soc: 82.000000 soh: 100
11:36:54.289 > [Pylontech] Received CAN message: 0x0356 - 02 14 D6 00 BC 00
11:36:54.294 > [Pylontech] voltage: 51.219997 current: 21.400000 temperature: 18.800001
11:36:54.300 > [Pylontech] Received CAN message: 0x035C - C0 00
11:36:54.333 > [Pylontech] chargeStatusBits: 1 1 0
11:36:54.333 > [Pylontech] Received CAN message: 0x035E - 50 59 4C 4F 4E 20 20 20
11:36:54.347 > [Pylontech] Manufacturer: PYLON   
11:36:54.577 > [PowerMeterHttpJson] New total: 2205.46
11:36:54.628 > RX Period End
11:36:54.641 > Success

Also die übliche Fehlersuche: Kabel gewechselt, beide Kabel durchgeklingelt und auf Belegung geprüft (Rx und Tx getauscht),
CAN / Iso Shield getauscht bzw. auf 2. Schnittstelle gewechselt, Victron hat alle Updates und Tx ist auf "normale Kommunikation" gestellt, verschiedene Einschaltreihenfolgen der Geräte getestet ==> NIX.
Daher das Teil hier https://github.com/datjan/esp8266-victron-mppt-solarchargecontroller zusammengeklöppelt und siehe da, der Victron lebt.
![victron](https://github.com/user-attachments/assets/f3c2654a-b4d2-40b8-a2d4-84b5547a7635)
Zumindest Produkt, Firmware und Seriennr. werden korrekt angezeigt. An ihm liegt es wohl nicht.

Hat jemand eine Idee was ich noch unternehmen könnte?

**Comment by SW-Niko** [comment]
Hallo @Matti07 ,
das schaut so aus als ob Daten ankommen aber die VE.Direct nichts damit anfangen kann.
Ich tippe auf einen Verdrahtungsfehler oder die Pin Konfiguration stimmt nicht.

Die Fusion V2 mit CAN / Iso Shield hat einen ADUM1201ARZ wenn ich das noch richtig im Kopf habe. Hast du auch GND und VCC zum Victron verbunden?

**Reply by Matti07** [reply]
> Hallo @Matti07 , das schaut so aus als ob Daten ankommen aber die VE.Direct nichts damit anfangen kann. Ich tippe auf einen Verdrahtungsfehler oder die Pin Konfiguration stimmt nicht.

Hallo @SW-Niko,
ja, genauso sieht es für mich auch aus. Wäre aber komisch. Das jetzige Kabel ist dasselbe wie bei dem 8266 Testgerät, nur die Breadboardstecker sind weg.
Durchgang prüfe ich mit Multimeter mit Krokoklemme und Nadel für die Steckerseite und direkt oben auf den Schrauben des CAN /Iso shields. Das wird dazu vom Fusionboard abgezogen.
Die jetzige Pinkonfiguration liefert immerhin den buffer overrun, sonst erscheint nicht mal diese Meldung.
GPIO11 	on SPI display header (CAN/Iso: Iso2 Ext TX / ESP RX)
GPIO12 	on SPI display header (CAN/Iso: Iso2 Ext RX / ESP TX)
für die 2. Schnittstelle.
https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md
 
> Die Fusion V2 mit CAN / Iso Shield hat einen ADUM1201ARZ wenn ich das noch richtig im Kopf habe. Hast du auch GND und VCC zum Victron verbunden?

GND ja, VCC nein.
Alle Bilder die ich gefunden habe zeigen nur 3 Verbindungen an, Rx, Tx und GND. Ist für mich auch logisch.
Die 5V Klemme am shield führt auch schon ca. 4,5V.
Bist Du Dir mit den verbundenen VCCs sicher?

**Reply by SW-Niko** [reply, marked as answer]
>Bist Du Dir mit den verbundenen VCCs sicher?

Ja, bin ich. Ich habe keine (Fusion V2 mit CAN / Iso Shield) aber einen ADUM1201ARZ und der macht eine galvanische Trennung und benötigt an beiden Seiten VCC und GND.

![grafik](https://github.com/user-attachments/assets/b579573f-52ab-4708-b7fb-75de7bfc024b)

Du kannst das auch hier nochmal nachlesen. https://opendtu-onbattery.net/hardware/vedirect/

>ja, genauso sieht es für mich auch aus. Wäre aber komisch. Das jetzige Kabel ist dasselbe wie bei dem 8266 Testgerät, nur die Breadboardstecker sind weg.

Das 8266 Testgerät verwendet einen Levelshifter. Und die benötigen nur selten 2 VCCs.

**Reply by schlimmchen** [reply]
Ich markiere das mal mutig als Antwort :wink:

@Matti07 Wenn du VCC deines Fusion I/O Shields nicht mit dem Victron VCC verbunden hast, dann hast du den ADUM1201 nicht verstanden. Der hat zwei völlig getrennte Spannungsdomänen und kodiert und dekodiert von der einen zur anderen Seite. Das ist aktive Logik und die braucht Strom. Du *musst* VCC anschließen.

**Comment by Matti07** [comment]
OK, schon mal vielen Dank für den Hinweis. Rückmeldung morgen.

**Comment by Matti07** [comment]
> Ja, bin ich. Ich habe keine (Fusion V2 mit CAN / Iso Shield) aber einen ADUM1201ARZ und der macht eine galvanische Trennung und benötigt an beiden Seiten VCC und GND.

Hallo @SW-Niko,
tja, das war mein Fehler, ich hatte die galv. Trennung vergessen ...

Btw., an der fehlenden VCC liegt es nicht bzw. nicht nur. Immer noch keine Verbindung.
Aber auch keinen "debug buffer overrun!" mehr.
Jetzt ist erstmal Pause bis Sa.
Rückmeldung folgt.

**Reply by SW-Niko** [reply]
Hmmm. 2 Fehler?
Dann kann es nur noch die Datenleitung sein. 
Der TX vom Victron muss mit Rx1 vom CAN / Iso Shield verbunden sein.
Der GND vom Victron muss mit GND1 vom CAN / Iso Shield verbunden sein.
Der VCC vom Victron muss mit VCC1 vom CAN / Iso Shield verbunden sein.

Oder alles mit Port 2.

**Reply by Matti07** [reply]
> Oder alles mit Port 2.

Genau so ist es z.Z.
Ich tippe momentan auf HW, werde das shield nochmal tauschen und notfalls noch ein Fusion-Board bestellen (brauche ich sowieso).

**Reply by SW-Niko** [reply]
Und in der config steht bei rx GPIO 11?

"victron": {
            "rx": 11,
}

**Reply by Matti07** [reply]
Fast ...
"victron": {
            "rx": -1,
            "tx": -1,
            "rx2": 11,
            "tx2": 12
         }

und das scheint oB auch gefressen zu haben:

Hardware-Einstellungen  - Victron - rx2	11	11

**Reply by SW-Niko** [reply]
Das könnte der Fehler sein ....
Ändere das mal folgendermaßen ab und probiere es nochmal.
"victron": {
"rx": 11,
"tx": -1,
"rx2": -1,
"tx2": -1
}

**Reply by Matti07** [reply]
Ändert leider nichts. Ich nutze den Anschluß der auf dem shield mit ...2 bezeichnet ist.
Rx auf 11 gemapped, Tx und Schnittstellen 2 und 3 mit -1 disabled.

**Reply by SW-Niko** [reply]
Jetzt gehen mir langsam die Ideen aus. 🤔 
Ich melde mich wenn mir noch was einfällt.

**Reply by Matti07** [reply]
Prima.
Niko, ich bedanke mich wirklich ausdrücklich für Deine Unterstützung und Deine Geduld bei Dir.

**Comment by Manos1966** [comment]
Was fuer ein VE.Direkt Kabel hast du?

**Reply by Matti07** [reply]
Ein originales Victronkabel mit 2 Steckern.
In der Mitte aufgetrennt, Kabel verlängert und damit dann an den shield.

**Reply by Manos1966** [reply]
> Ein originales Victronkabel

Aha! 👍 Aber das originale Victronkabel ist ein CROSSOVER Kabel 😉 

TX geht (im KABEL) auf RX

Wir hatten das Problem vor vier Tagen... ich suche 

Hier: https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/1236#discussioncomment-10705832

**Reply by Matti07** [reply]
OK, Danke für den Hinweis, werde ich wohl morgen hoffentlich prüfen können.
Allerdings hat mein kleines Testgerät (s.o.) mit genau dem jetzigen Kabel funktioniert.
Durchgang prüfe ich mit Multimeter mit Krokoklemme und Nadel für die Steckerseite, also quasi im Stecker, und direkt oben auf den Schrauben des CAN /Iso shields. Das wird dazu vom Fusionboard abgezogen.
Ich denke, da wäre mir eine Kreuzung aufgefallen.
Aber ich teste das trotzdem.

**Reply by Manos1966** [reply]
genau darum "dein kleines Testgerät mit genau dem jetzigen Kabel funktioniert" bin ich drauf gekommen, dass du ein Original Victron Kabel benutzt haben musst 😉

**Reply by schlimmchen** [reply]
> Kabel verlängert

Wie viel länger? Da wird UART-TTL gesprochen, das funktioniert nur auf vergleichweise kurzen Leitungen. Wenn du da fünf Meter Kabel dazwischen gefummelt hast, dann ist es sehr gut möglich, dass das Signal am anderen Ende schon nicht mehr zu dekodieren ist. Außerdem ist Grounding/Shielding dann wichtig, etc.

Die [Doku](https://opendtu-onbattery.net/hardware/vedirect/) hast du hoffentlich schon auswendig gelernt?

**Reply by Matti07** [reply]
Hallo @schlimmchen,
die Gesamtlänge beträgt ca. 80cm.
Ich habe aber einen Verdacht: verlängert habe ich das Kabel mit Pfuscherperlen (Schwachstromverbindern).
Beim CAN Bus funktioniert das einwandfrei und auch sonst habe ich mit den Dingern noch nie Probleme gehabt.
Ich werde spätestens morgen ein neues Kabel basteln, diesmal aber verlötet oder gepresst.
Ich melde mich sofort bei neuen Erkenntnissen.

**Reply by SW-Niko** [reply]
Du kannst ja mal folgenden Check machen. 
- Das Kabel nicht mit den Victron verbinden
- Stattdessen am Kabel/Stecker (Seite zum Victron) Tx mit VCC verbinden. Dann müssen am GPIO11 ca. 3V3 anliegen.
- Dann die Gegenprobe. Tx mit GND verbinden.  Dann muss am GPIO11 ca. 0V anliegen.

**Comment by Matti07** [comment]
Moin zusammen,
es geht für mich, wie oben angekündigt, leider erst am Sa. weiter.
Ich bekomme Stress mit meiner Heimleiterin ...

**Comment by Matti07** [comment]
Hallo @SW-Niko , @schlimmchen und @Manos1966 ,
vielen Dank für eure Tipps und Hinweise, sie haben leider nichts geändert.
Es ist wohl tatsächlich das Fusion Board defekt.
Seit gestern kein WLAN mehr und seit heute WLAN reproduzierbar max. 1,5 min.
Am WLAN habe ich seit Wochen nichts gemacht und an die Antennen und Buchsen kommt man nicht so einfach ran.
Laut meiner Fritte ist die Verbindung perfekt, leider kann ich oB nicht mal anpingen.
Auch ein Zugriff auf den eigenen AP ist nicht möglich.
Die Steuerung funktioniert anscheinend noch (USB log-datei), reagiert aber nicht mehr auf die webap.
Aber egal, ich tausche jetzt erst mal das Board.
Abschlußmeldung folgt.
